pkg/controller/daemon/daemoncontroller.go

```diff
@@ -818,9 +822,8 @@ func (dsc *DaemonSetsController) nodeShouldRunDaemonPod(node *v1.Node, ds *exten
 			// unexpected
 			case
 				predicates.ErrPodAffinityNotMatch,
-				predicates.ErrServiceAffinityViolated,
-				predicates.ErrTaintsTolerationsNotMatch:
-				return false, false, false, fmt.Errorf("unexpected reason: GeneralPredicates should not return reason %s", reason.GetReason())
+				predicates.ErrServiceAffinityViolated:
+				return false, false, false, fmt.Errorf("unexpected reason: deamonSetPredicates should not return reason %s", reason.GetReason())
```

Comment:
Just as a point of comparison, the `default` case below has extra logging.  If the "unexpected" reason should not happen, I think it's worth a warning message, especially because V(4) messages don't show in a production cluster (V(2) by default). Of course, this should not cause any spam in the logs.